### PR TITLE
Feature 683: Fixed server status check and made login async

### LINKED ISSUE #683: The function checkServerConnection always throws an error
Whenever a user tries to login a screen pops up with a message telling that they cannot connect to the server. This is caused by a http call in the login_screen, which always returns with a satuscode 404 (not found). Then it is checked against the value 200(success), and an exception is thrown. This does not prevent the user from logging in though, as the login is async and this function seems to always be called.

1. Set a breakpoint in login_screen on the line `if (loginResponse.statusCode == 200) {`
2. login on the weekplanner with debugging enabled
3. Observe that the value of `loginResponse.statusCode` is 404

**Expected behavior**
Either the response should be 200 when there is a connection. and if there isnt a connection a user should not be able to log in.

**Actual behavior**
No matter what, the response code is 404 and the user can still log in as, there is a connection

**Environment (please complete the following information):**
- OS: Android
- Emulator: yes
- Phone: Pixel 2
- APK Version: 30

### PR DESCRIPTION
**Pull request by AdrianPlesner**
Closes #683  
The check server status method did not request the correct api-endpoint, always resulting in 404 route not found. While fixing this, we also discovered that the login was completely backwards and not actually async. This, being related to the issue,  has been fixed as well.

### CODE REVIEW

**Comment by Determinatoro** [comment]
Please use the following checklist, @JeppeKLau. Feel free to check it off as you go along.

**Assets**
- [x] The ``environments.json`` file has the following default values:
```
{
  "SERVER_HOST": "https://srv.giraf.cs.aau.dk/DEV/API",
  "DEBUG": true,
  "USERNAME": "Graatand",
  "PASSWORD": "password"
}
```

**Code Design** 

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems     
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability** 

- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [x] The code is simple and readable. i.e. it contains no \"hack\", or obscure solution   

**Code Maintainability**

- [ ] Has all the added code been properly commented

Only check one of the items below:

- [ ] The functionality does **not** need to be covered by tests
- [x] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [x] All functionality is covered by tests
- [x] The tests has sensible names
- [x] By reading the tests you know what they cover
- [x] The tests cover sensible cases. Both happy paths and exception paths
- [x] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

**Functionality**

For this part, open the app, and test:
- [x] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation

**Comment by Determinatoro** [comment]
Please use the following checklist, @Riasalit. Feel free to check it off as you go along.

**Assets**
- [x] The ``environments.json`` file has the following default values:
```
{
  "SERVER_HOST": "https://srv.giraf.cs.aau.dk/DEV/API",
  "DEBUG": true,
  "USERNAME": "Graatand",
  "PASSWORD": "password"
}
```

**Code Design** 

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems     
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability** 

- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [x] The code is simple and readable. i.e. it contains no \"hack\", or obscure solution   

**Code Maintainability**

- [x] Has all the added code been properly commented

Only check one of the items below:

- [ ] The functionality does **not** need to be covered by tests
- [x] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [x] All functionality is covered by tests
- [x] The tests has sensible names
- [x] By reading the tests you know what they cover
- [x] The tests cover sensible cases. Both happy paths and exception paths
- [x] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

**Functionality**

For this part, open the app, and test:
- [x] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation

File: lib/blocs/auth_bloc.dart [hunk 1]
```diff
@@ -47,17 +50,30 @@ class AuthBloc extends BlocBase {
         loggedInUsername = username;
         setMode(WeekplanMode.guardian);
       }
+      completer.complete();
+    }).onError((Object error){
+      completer.completeError(error);
     });
+    // ignore: always_specify_types
```
Comment (Riasalit) [inline lib/blocs/auth_bloc.dart]: Hvorfor ikke bare specificere en type?

**Review by Riasalit — COMMENTED** [review]
image

Altså ved ikke om det her skyldes den her branch... men kan ikke logge ind

**Comment by AdrianPlesner** [comment]
@Riasalit Jeg har brug for at se mere af loggen fra dit api for at kunne identificere hvad der er galt, men mit bud er at det nok har noget at gøre med den kombination af web-api og api_client du bruger. Det er testet og virker med den nyeste version af develop af begge.

**Comment by AdrianPlesner** [comment]
Account controller I web-api er fornyligt ændret fra v1 til v2 så hvis man ikke får opdateret både web-api og api_client kan man ikke logge ind da end-points så ikke passer sammen.

File: lib/blocs/auth_bloc.dart [hunk 2]
```diff
@@ -47,17 +50,30 @@ class AuthBloc extends BlocBase {
         loggedInUsername = username;
         setMode(WeekplanMode.guardian);
       }
+      completer.complete();
+    }).onError((Object error){
+      completer.complete();
     });
```
Comment (JeppeKLau) [inline lib/blocs/auth_bloc.dart]: Umiddelbart vil jeg mene at det ikke er smart bare at gennemtvinge en færdiggørelse ved en fejl.

File: lib/blocs/auth_bloc.dart [hunk 3]
```diff
@@ -47,17 +50,30 @@ class AuthBloc extends BlocBase {
...
     _api.account.login(username, password).listen((bool status) {
       if (status) {
           _loginAttempt.add(status);
           setMode(WeekplanMode.guardian);
         }
-    });
+      completer.complete();
+    }).onError((Object error) {
+      completer.completeError(error);
+    } );
```
Comment (JeppeKLau) [inline lib/blocs/auth_bloc.dart:73]: Igen med at gennemtvinge gennemførsel på en fejl.

File: lib/screens/login_screen.dart [hunk 4]
```diff
@@ -260,7 +262,7 @@ class LoginScreenState extends State<LoginScreen> {
   /// Function to test connection to server,
   /// it both checks for DEV API connection and to PROD API connection
   Future<bool> checkServerConnection() async {
-    final String loginUrl = environment.getVar<String>('SERVER_HOST');
+    final String loginUrl = environment.getVar<String>('SERVER_HOST') + '/v1/Status';
```
Comment (JeppeKLau) [inline lib/screens/login_screen.dart]: Jeg kan forestille mig at man kan tilgå statusen på en anden måde, fx. gennem api klienten. Ellers burde det tilføjes.

Udover dette, giver navnet på variablen ikke mening da det ikke er på det endpoint vi logger ind.

**Review by JeppeKLau — CHANGES_REQUESTED** [review]
Error handling er en stor mangel.

Udover kommentarene, kunne jeg forestille mig der var smartere måder at tjekke om vi har forbindelse til internettet, end blot at pinge Googles url. Ved godt det ikke er direkte relateret til issuet, men jeg vil nu stadig gerne se det ændret.

Umiddelbart vil jeg også sige at jeg ikke har hørt om den der `Completer` som bliver brugt. Kunne forestille mig at samme funktionalitet kan opnås ved bare at have nogle gode asynkrone funktioner, der bliver ventet på de rigtige steder.

Se også manglende punkter på checklisten.

File: lib/blocs/auth_bloc.dart (hunk 3 above)
Comment (AdrianPlesner) [reply lib/blocs/auth_bloc.dart:73]: Men .completeError(error) melder jo nettop færdig med fejl og sender fejlen videre til caller [completeError dokumentation](https://api.dart.dev/stable/2.10.2/dart-async/Completer/completeError.html). Hvis du ikke synes det er tilstrækkeligt kunne jeg godt tænke mig at du kommer med et konkret forslag til hvad der så skal ske.

Comment (JeppeKLau) [reply lib/blocs/auth_bloc.dart:73]: Oh. Jeg har læst forkert, havde set det som at der blev kaldt `complete()`.

**Comment by Riasalit** [comment]
```Welcome to Giraf REST Server.
        No program arguments were found - running in default configuration.
        Run with --list to list options
info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[0]
      User profile is available. Using 'C:\Users\danie\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
info: Microsoft.EntityFrameworkCore.Infrastructure[10403]
      Entity Framework Core 3.1.8 initialized 'GirafDbContext' using provider 'Pomelo.EntityFrameworkCore.MySql' with options: None
info: Microsoft.EntityFrameworkCore.Database.Command[20101]
      Executed DbCommand (35ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT EXISTS (
          SELECT 1
          FROM `AspNetRoles` AS `a`)
Hosting environment: Development
Content root path: C:\Users\danie\Documents\GitHub\web-api\GirafRest
Now listening on: http://[::]:5000
Application started. Press Ctrl+C to shut down.
info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/1.1 POST http://10.0.2.2:5000/v1/Account/login application/json; charset=utf-8 45
info: Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler[2]
      Successfully validated the token.
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[3]
      Route matched with {action = "StatusCodeEndpoint", controller = "Error"}. Executing controller action with signature Microsoft.AspNetCore.Mvc.ActionResult StatusCodeEndpoint(Int32) on controller GirafRest.Controllers.ErrorController (GirafRest).
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[1]
      Executing action method GirafRest.Controllers.ErrorController.StatusCodeEndpoint (GirafRest) - Validation state: Valid
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[2]
      Executed action method GirafRest.Controllers.ErrorController.StatusCodeEndpoint (GirafRest), returned result Microsoft.AspNetCore.Mvc.NotFoundObjectResult in 0.6537ms.
info: Microsoft.EntityFrameworkCore.Infrastructure[10403]
      Entity Framework Core 3.1.8 initialized 'GirafDbContext' using provider 'Pomelo.EntityFrameworkCore.MySql' with options: None
info: Microsoft.EntityFrameworkCore.Database.Command[20101]
      Executed DbCommand (45ms) [Parameters=[@__userId_0='?' (Size = 255)], CommandType='Text', CommandTimeout='30']
      SELECT `a`.`Id`, `a`.`AccessFailedCount`, `a`.`ConcurrencyStamp`, `a`.`DepartmentKey`, `a`.`DisplayName`, `a`.`Email`, `a`.`EmailConfirmed`, `a`.`IsDepartment`, `a`.`LockoutEnabled`, `a`.`LockoutEnd`, `a`.`NormalizedEmail`, `a`.`NormalizedUserName`, `a`.`PasswordHash`, `a`.`PhoneNumber`, `a`.`PhoneNumberConfirmed`, `a`.`SecurityStamp`, `a`.`SettingsKey`, `a`.`TwoFactorEnabled`, `a`.`UserIcon`, `a`.`UserName`
      FROM `AspNetUsers` AS `a`
      WHERE `a`.`Id` = @__userId_0
      LIMIT 1
info: Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor[1]
      Executing ObjectResult, writing value of type 'GirafRest.Models.Responses.ErrorResponse'.
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[2]
      Executed action GirafRest.Controllers.ErrorController.StatusCodeEndpoint (GirafRest) in 515.0133ms
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished in 1039.9936ms 404 application/json; charset=utf-8
info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/1.1 GET http://10.0.2.2:5000/v1/Status  0
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[3]
      Route matched with {action = "Status", controller = "Status"}. Executing controller action with signature Microsoft.AspNetCore.Mvc.ActionResult Status() on controller GirafRest.Controllers.StatusController (GirafRest).
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[1]
      Executing action method GirafRest.Controllers.StatusController.Status (GirafRest) - Validation state: Valid
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[2]
      Executed action method GirafRest.Controllers.StatusController.Status (GirafRest), returned result Microsoft.AspNetCore.Mvc.OkObjectResult in 0.2426ms.
info: Microsoft.EntityFrameworkCore.Infrastructure[10403]
      Entity Framework Core 3.1.8 initialized 'GirafDbContext' using provider 'Pomelo.EntityFrameworkCore.MySql' with options: None
info: Microsoft.EntityFrameworkCore.Database.Command[20101]
      Executed DbCommand (1ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
      SELECT `a`.`Id`, `a`.`AccessFailedCount`, `a`.`ConcurrencyStamp`, `a`.`DepartmentKey`, `a`.`DisplayName`, `a`.`Email`, `a`.`EmailConfirmed`, `a`.`IsDepartment`, `a`.`LockoutEnabled`, `a`.`LockoutEnd`, `a`.`NormalizedEmail`, `a`.`NormalizedUserName`, `a`.`PasswordHash`, `a`.`PhoneNumber`, `a`.`PhoneNumberConfirmed`, `a`.`SecurityStamp`, `a`.`SettingsKey`, `a`.`TwoFactorEnabled`, `a`.`UserIcon`, `a`.`UserName`
      FROM `AspNetUsers` AS `a`
      WHERE FALSE = TRUE
      LIMIT 1
info: Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor[1]
      Executing ObjectResult, writing value of type 'GirafRest.Models.Responses.SuccessResponse'.
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[2]
      Executed action GirafRest.Controllers.StatusController.Status (GirafRest) in 41.8183ms
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished in 52.5691ms 200 application/json; charset=utf-8 ```

**Comment by Riasalit** [comment]
Soooo flutter simply refuses to get the latest version of the api-client

**Comment by AdrianPlesner** [comment]
> Soooo flutter simply refuses to get the latest version of the api-client

Yup det virker til at være problemet

**Comment by Riasalit** [comment]
image

We in bois.
"flutter packages upgrade" og så slet den forkerte cachede versio...... igen igen... virkede

**Comment by JeppeKLau** [comment]
> 
> 
> Soooo flutter simply refuses to get the latest version of the api-client

Jeg fik det til at virke ved at fjerne `api_client` fra `pubspec.yaml`, køre `flutter pub get`, tilføje den tilbage i `pubspec.yaml` og køre `flutter pub get` igen - ser ud til at `pubspec.lock` har en forkert `resolved-ref` for `api_client`

**Comment by AdrianPlesner** [comment]
> > Soooo flutter simply refuses to get the latest version of the api-client
> 
> Jeg fik det til at virke ved at fjerne `api_client` fra `pubspec.yaml`, køre `flutter pub get`, tilføje den tilbage i `pubspec.yaml` og køre `flutter pub get` igen - ser ud til at `pubspec.lock` har en forkert `resolved-ref` for `api_client`

Jeg bruger samme methode

**Comment by Riasalit** [comment]
> 
> 
> > > Soooo flutter simply refuses to get the latest version of the api-client
> > 
> > 
> > Jeg fik det til at virke ved at fjerne `api_client` fra `pubspec.yaml`, køre `flutter pub get`, tilføje den tilbage i `pubspec.yaml` og køre `flutter pub get` igen - ser ud til at `pubspec.lock` har en forkert `resolved-ref` for `api_client`
> 
> Jeg bruger samme methode

prøvede det. men det gjorde intet for mig

**Review by Riasalit — DISMISSED** [review]

**Review by Riasalit — APPROVED** [review]

**Review by Riasalit — CHANGES_REQUESTED** [review]
CI actions failer

**Review by Riasalit — CHANGES_REQUESTED** [review]
Vil gerne have at der står hvad funktionen gør

**Comment by AdrianPlesner** [comment]
> Vil gerne have at der står hvad funktionen gør

Ups. 😄

**Review by Riasalit — DISMISSED** [review]

File: lib/screens/login_screen.dart [hunk 5]
```diff
@@ -45,49 +44,53 @@ class LoginScreenState extends State<LoginScreen> {
...
-            // Checking username/password
-            if (!loginStatus) {
-              creatingNotifyDialog('Forkert brugernavn og/eller adgangskode.',
-                  'WrongUsernameOrPassword');
-            }
+      else if(error is SocketException){
+        checkInternetConnection().then((bool hasInternetConnection) {
+          if (hasInternetConnection) {
+            // Checking server connection, if true check username/password
+            checkServerConnection().then((bool hasServerConnection) {
```
Comment (JeppeKLau) [inline lib/screens/login_screen.dart]: Jeg tænker at hvis man får en `SocketException`, så må der være noget galt med enten internet forbindelsen eller forbindelsen til serveren. Så jeg tænker det måske er fint nok bare at kontrollere forbindelsen til serveren, da det ikke er nødvendigt at tjekke forbindelsen til internettet hvis man alligevel får en `SocketException`.

File: test/screens/login_screen_test.dart [hunk 6]
```diff
@@ -33,7 +33,6 @@ class MockLoginScreenState extends LoginScreenState {
   }
 
   /// This is the callback method of the loading spinner to show the dialog
-  @override
   void showNotifyDialog() {
     // Checking username/password
```
Comment (JeppeKLau) [inline test/screens/login_screen_test.dart]: Når du har fjernet funktionen i koden, ser jeg ingen grund til at bringe funktionen tilbage til testen. Hvis testen er afhængig af funktionen, må testen skrives om så den ikke er afhængig af funktionen.

File: lib/blocs/auth_bloc.dart [hunk 7]
```diff
@@ -47,17 +50,43 @@ class AuthBloc extends BlocBase {
         loggedInUsername = username;
         setMode(WeekplanMode.guardian);
       }
+      completer.complete();
+    }).onError((Object error){
+      completer.completeError(error);
     });
+    // ignore: always_specify_types
```
Comment (JeppeKLau) [inline lib/blocs/auth_bloc.dart]: Kan typen ikke tilføjes et sted eller er det bare linteren der er dum?

**Review by JeppeKLau — CHANGES_REQUESTED** [review]
Se kommentarer :)

File: lib/blocs/auth_bloc.dart (hunk 7 above)
Comment (AdrianPlesner) [reply lib/blocs/auth_bloc.dart]: Det er en rigtig dum linter 🙂

File: lib/screens/login_screen.dart (hunk 5 above)
Comment (AdrianPlesner) [reply lib/screens/login_screen.dart]: Det sørger jo bare for at håndtere fejl vi ikke regner med at der kommer. Det er bare for en sikkerheds skyld. Jeg tænker heller ikke at der kan være andet end manglede forbindelse til enten server eller internet, men man kan vel ikke være helt sikker. 🤷

Comment (JeppeKLau) [reply lib/screens/login_screen.dart]: Det er bare et spildt tjek. Desuden, som jeg også sagde før, er det virkelig ikke smart at tjekke om man har internet ved at pinge Google. Der er ikke noget der siger at deres service ikke kunne ske at gå ned på et tidpunkt i fremtiden.

File: test/screens/login_screen_test.dart (hunk 6 above)
Comment (AdrianPlesner) [reply test/screens/login_screen_test.dart]: Fixed

File: lib/screens/login_screen.dart (hunk 5 above)
Comment (AdrianPlesner) [reply lib/screens/login_screen.dart]: Det er da en meget standard måde at gøre det på, og så er det cross platform.

File: lib/blocs/auth_bloc.dart (hunk 7 above)
Comment (Riasalit) [reply lib/blocs/auth_bloc.dart]: er det ikke bare `Future.wait(<Future<void>>[completer.future]);`?

Comment (JeppeKLau) [reply lib/blocs/auth_bloc.dart]: Det er bare `Future.wait(<Future<void>>[completer.future]);`

File: lib/screens/login_screen.dart (hunk 5 above)
Comment (JeppeKLau) [reply lib/screens/login_screen.dart]: Der er [det her](https://pub.dev/packages/connectivity#-readme-tab-) plugin som jeg kan se ville kunne short circuit at tjekke om man har internet, inden man begynder på at pinge rundt på nettet.
Dertil vil jeg så nævne at der også findes [det her](https://medium.com/@anilpandey071999/how-to-check-the-internet-connection-in-the-flutter-app-56b8be1fd057) plugin der automatisk pinger en række af hjemmesider og ikke blot en enkelt som vi gør her. Det giver en mindre chance for fejl da de alle skal misse før der konkluderes at man ingen forbindelse har. 

De to plugins i sammenhæng kunne jeg se gav en god sikkerhed for at tjekke om man har internet eller ej.

File: lib/blocs/auth_bloc.dart (hunk 7 above)
Comment (AdrianPlesner) [reply lib/blocs/auth_bloc.dart]: Jeg overgiver mig på den front, jeg synes bare at jeg havde prøvet, men min compiler var vidst forvirret. Det virker til at virke nu. 😄

File: lib/screens/login_screen.dart (hunk 5 above)
Comment (AdrianPlesner) [reply lib/screens/login_screen.dart]: Oki doki så, hvis du insisterer.

File: lib/blocs/auth_bloc.dart [hunk 8]
```diff
@@ -81,11 +83,21 @@ class AuthBloc extends BlocBase {
     _api.status.status().listen((bool status) {
       completer.complete(status);
     }).onError((Object error){
-      completer.completeError(error);
+      completer.complete(false);
```
Comment (JeppeKLau) [inline lib/blocs/auth_bloc.dart]: Var det ikke den der skulle give en error tilbage når der skete en fejl?

File: lib/screens/login_screen.dart [hunk 9]
```diff
@@ -106,6 +107,11 @@ class LoginScreenState extends State<LoginScreen> {
               title: 'Fejl', description: description, key: Key(key));
         });
   }
+  /// Create an unknown error dialog
+  void unknownErrorDialog(String key){
+    creatingNotifyDialog('Der skete en ukendt fejl, prøv igen eller '
+        'kontakt en administrator', 'key');
+  }
```
Comment (JeppeKLau) [inline lib/screens/login_screen.dart:114]: Jeg kan godt se ideen, men virker bare lidt fjollet når den kun bruges tre gange, hvor man bare kunne indsætte `creatingNotifyDialog` med den passende key

**Review by JeppeKLau — CHANGES_REQUESTED** [review]
Vi er der næsten Adrian. Vil tilføje at det er dejligt at se du har rykket business logic om i BLOCen frem for at have det i screenen

File: lib/screens/login_screen.dart (hunk 9 above)
Comment (AdrianPlesner) [reply lib/screens/login_screen.dart:114]: Jah, jeg synes 3 gange var for meget

File: lib/blocs/auth_bloc.dart (hunk 8 above)
Comment (AdrianPlesner) [reply lib/blocs/auth_bloc.dart]: Joh, men det er egentligt nok at håndter fejler her. At smide fejler videre giver bare endnu et unødvendigt led med onError -> createErrorDialog, og i sidste ende giver det samme resultat

File: lib/screens/login_screen.dart (hunk 5 above)
Comment (AdrianPlesner) [reply lib/screens/login_screen.dart]: Vi kan ikke bruge connectivity pakken, den giver build errors på android, jeg håber du kan nøjes med den anden.

**Review by Riasalit — APPROVED** [review]

File: lib/screens/login_screen.dart (hunk 5 above)
Comment (JeppeKLau) [reply lib/screens/login_screen.dart]: Det må være nok så - tænkte bare der måske var et eller andet indbygget, men det her må være nok

**Review by JeppeKLau — APPROVED** [review]

**Pull request merged**